pkg/driver/node_server.go

```diff
@@ -708,89 +708,36 @@ func (driver *Driver) NodePublishVolume(ctx context.Context, request *csi.NodePu
 
-		var secrets map[string]string
-
-		// Get the secrets from the volume attributes if secrets are unspecified or nil
-		if request.Secrets != nil {
-			log.Tracef("Referencing secrets from the request for %s", request.VolumeId)
-			secrets = request.Secrets
-		} else {
-			secretName := request.VolumeContext[secretNameKey]
-			secretNamespace := request.VolumeContext[secretNamespaceKey]
-
-			if secretName == "" || secretNamespace == "" {
-				errMsg := fmt.Sprintf("Failed to node publish ephemeral volume %s, Missing secrets in the request", request.VolumeId)
-				log.Errorf("err: %s", errMsg)
-				return nil, status.Error(codes.InvalidArgument, errMsg)
-			}
-
-			log.Tracef("Referencing secrets from volume attributes for %s, secretRef: %s/%s",
-				request.VolumeId, secretNamespace, secretName)
-			var err error
-			secrets, err = driver.flavor.GetCredentialsFromSecret(secretName, secretNamespace)
-			if err != nil {
-				errMsg := fmt.Sprintf("Failed to node publish ephemeral volume %s, err: %s", request.VolumeId, err.Error())
-				log.Errorf("err: %s", errMsg)
-				return nil, status.Error(codes.InvalidArgument, errMsg)
-			}
-		}
-
-		// Get/Construct ephemeral volume name
-		podName := request.VolumeContext[csiEphemeralPodName]
-		ephemeralVolName := getEphemeralVolName(
+		// Construct ephemeral volume name from the given volume handle and podName
+		volumeName := getEphemeralVolName(
 			podName,          // POD Name
 			request.VolumeId, // Ephemeral Volume Handle
 		)
-		log.Infof("Provisioning ephemeral volume %s (volumeHandle: %s) for pod %s",
-			ephemeralVolName, request.VolumeId, podName)
-
-		// Node publish
-		{
-			// serialize ephemeral NodePublish() requests
-			ephemeralPublishLock.Lock()
-			defer ephemeralPublishLock.Unlock()
-
-			err = driver.nodePublishEphemeralVolume(
-				request.VolumeId, // Ephemeral CSI Volume Handle
-				ephemeralVolName,
-				stagingTargetPath,
-				request.TargetPath,
-				request.VolumeCapability,
-				secrets,
-				request.Readonly,
-				request.VolumeContext,
-			)
-			if err != nil {
-				log.Errorf("Failed to node publish ephemeral volume %s, err: %s", request.VolumeId, err.Error())
-
-				rbErr := driver.retryRollbackEphemeralVolume(
-					request.VolumeId, // Ephemeral CSI Volume Handle
-					ephemeralVolName,
-					request.Secrets,
-					stagingTargetPath,
-					request.TargetPath,
-				)
-				if rbErr != nil {
-					log.Errorf("Failed to cleanup/rollback ephemeral volume %s, err: %s",
-						request.VolumeId, rbErr.Error())
-					// Returning both original error + rollback error
-					return nil, status.Error(codes.Internal,
-						fmt.Sprintf("%s, %s", err.Error(), rbErr.Error()))
-				}
-				log.Infof("Cleanup/Rollback of ephemeral volume %s successful", request.VolumeId)
-				return nil, err
-			}
+		err := driver.nodePublishEphemeralVolume(
+			request.VolumeId, // Ephemeral CSI Volume Handle
+			volumeName,
+			stagingTargetPath,
+			request.TargetPath,
```

Comment:
may be better to just pass entire request object itself?

Reply:
Some attributes like 'volumeName' from 'volumeID' and 'stagingTargetPath' from 'targetPath' are derived/constructed from the request object before passing down to this function and other functions. so, all these fields intact and common they are passed separately to avoid another interpretation with that called function.

Reply:
ok..thanks.